net/portmapper/portmapper.go

```diff
@@ -113,7 +115,7 @@ func (c *Client) HaveMapping() bool {
 //
 // All fields are immutable once created.
 type pmpMapping struct {
-	gw         netaddr.IP
+	gw         netaddr.IPPort
```

Comment:
I guess this also makes sense considering a mapping's gateway won't just be the IP but also the port to access it, the change feels a bit odd but seems more right